File: 017_Bubble_Sort/Bubble_Sort.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> bubbleSort(std::vector<int>& arr) {
        int n = arr.size();
        for (int i = 0; i < n - 1; ++i) {
            for (int j = 0; j < n - i - 1; ++j) {
                if (arr[j] > arr[j + 1]) {
                    std::swap(arr[j], arr[j + 1]);
                }
            }
        }
        return arr;
    }
};

int main() {
    // Example usage
    Solution solution;
    std::vector<int> exampleArray = {64, 34, 25, 12, 22, 11, 90};
    std::vector<int> sortedArray = solution.bubbleSort(exampleArray);

    // Print original and sorted arrays
    std::cout << "Original array:";
    for (int num : exampleArray) {
        std::cout << " " << num;
    }
    std::cout << std::endl;

    std::cout << "Sorted array:";
    for (int num : sortedArray) {
        std::cout << " " << num;
    }
    std::cout << std::endl;

    return 0;
}

// command to run code
// g++ -std=c++17 017_Bubble_Sort/Bubble_Sort.cpp

File: 017_Bubble_Sort/Bubble_Sort.py
class Solution:
    def bubble_sort(self, arr):
        n = len(arr)
        for i in range(n - 1):
            swapped = False
            for j in range(n - i - 1):
                if arr[j] > arr[j + 1]:
                    arr[j], arr[j + 1] = arr[j + 1], arr[j]
                    swapped = True
            if not swapped:
                break
        return arr

# Example usage
solution = Solution()
example_array = [64, 34, 25, 12, 22, 11, 90]
sorted_array = solution.bubble_sort(example_array)

# Print original and sorted arrays
print("Original array:", *example_array)
print("Sorted array:", *sorted_array)
